server/src/modules/mail/mail.service.ts

```diff
@@ -1,18 +1,17 @@
-import { createTransport, Transporter } from 'nodemailer'
-import SMTPTransport from 'nodemailer/lib/smtp-transport'
+import { Transporter } from 'nodemailer'
 import { renderLoginOtpBody } from './mail.util'
 
 export class MailService {
   private mailFromEmail: string
-  private transport: Transporter
+  private transport: Pick<Transporter, 'sendMail'>
   constructor({
```

Comment:
Theres syntactic sugar for this, credit to @mantariksh.
https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties